evalml/tests/component_tests/test_simple_imputer.py

```diff
@@ -549,10 +551,16 @@ def test_simple_imputer_ignores_natural_language(
         if numeric_impute_strategy == "mean" and has_nan == "has_nan":
             ans = X_df.mean()
             ans["natural language col"] = pd.NA
+            X_df = X_df.astype(
+                {"int col": float},
+            )  # Convert to float as the imputer will do this as we're requesting the mean
             X_df.iloc[-1, :] = ans
         elif numeric_impute_strategy == "median" and has_nan == "has_nan":
             ans = X_df.median()
             ans["natural language col"] = pd.NA
+            X_df = X_df.astype(
+                {"int col": float},
+            )  # Convert to float as the imputer will do this as we're requesting the mean
```

Comment:
We shouldn't need this casting, median is fine staying as an integer type!

Reply:
I believe `median` can end up imputing a float between integer values so to play it safe, it's being cast as such in the `SimpleImputer` for both `mean` and `median`